dandi/models.py

```diff
@@ -116,6 +116,185 @@ def diff_models(model1, model2):
+            attributes.setdefault("fundingReferences", []).append(dict_fund)
+            # if no more roles, it shouldn't be added to creators or contributors
+            contr_el.roleName.remove(RoleType("dandi:Sponsor"))
+            if not contr_el.roleName:
+                continue
+
+        contr_dict = {
+            "name": contr_el.name,
+            "contributorName": contr_el.name,
+            "schemeURI": "orcid.org",
+        }
+        if isinstance(contr_el, Person):
+            contr_dict["nameType"] = "Personal"
+            if len(contr_el.name.split(",")) == 2:
+                contr_dict["familyName"], contr_dict["givenName"] = contr_el.name.split(
+                    ","
+                )
+            if getattr(contr_el, "affiliation"):
+                contr_dict["affiliation"] = [
+                    {"name": el.name} for el in contr_el.affiliation
+                ]
+            else:
+                contr_dict["affiliation"] = []
+        elif isinstance(contr_el, Organization):
+            contr_dict["nameType"] = "Organizational"
+
+        if RoleType("dandi:Author") in getattr(contr_el, "roleName"):
+            create_dict = deepcopy(contr_dict)
+            create_dict["creatorName"] = create_dict.pop("contributorName")
+            creators.append(create_dict)
+            contr_el.roleName.remove(RoleType("dandi:Author"))
+            # if no more roles, it shouldn't be added to contributors
+            if not contr_el.roleName:
+                continue
+
+        if getattr(contr_el, "roleName"):
+            contr_all = [
+                el.name for el in contr_el.roleName if el.name in DATACITE_CONTRTYPE
+            ]
+            if contr_all:
+                contr_dict["contributorType"] = contr_all[0]
+            else:
+                contr_dict["contributorType"] = "Other"
+        contributors.append(contr_dict)
+
+    # if there are no creators, the first contributor is also treated as the creator
+    if not creators and contributors:
+        creators = [deepcopy(contributors[0])]
+        creators[0]["creatorName"] = creators[0].pop("contributorName")
+        creators[0].pop("contributorType")
+
+    attributes["contributors"] = contributors
+    attributes["creators"] = creators
+
+    if getattr(meta, "relatedResource"):
+        attributes["relatedIdentifiers"] = []
+        for rel_el in meta.relatedResource:
+            ident = rel_el.identifier.split(":")
+            if len(ident) == 2:
+                ident_tp, ident_nr = ident
+                if ident_tp.lower() in DATACITE_MAP:
+                    ident_tp = DATACITE_MAP[ident_tp.lower()]
+                else:
+                    raise ValueError(
+                        f"identifier has to be from the list: {DATACITE_IDENTYPE}, "
+                        f"but {ident_tp} provided"
+                    )
+            else:
+                raise Exception("identifier is expected to be type:number")
```

Comment:
seems to be also of `ValueError` kind and we better inform the value on which we choked:

```suggestion
                raise ValueError(f"identifier is expected to be type:number, got {rel_el.identifier}")
```

Reply:
Let  me just proceed with this invaluable contribution of mine -- hopefully it doesn't break anything